test/allowlist_for_publicAPI.json

```diff
@@ -2388,12 +2388,10 @@
   ],
   "torch.utils.data.datapipes.gen_pyi": [
     "Any",
-    "Dict",
-    "List",
-    "Set",
-    "Tuple",
     "Union",
-    "defaultdict"
+    "defaultdict",
+    "Path",
```

Comment:
These are implicit reexports, we can remove them with a __all__ or prefixing them with _